install/swiftly-install.sh

```diff
@@ -71,9 +115,57 @@ bold () {
     echo "$(tput bold)$1$(tput sgr0)"
 }
 
+# Fetch the list of required system dependencies from the apple/swift-docker
+# repository and attempt to install them using the system's package manager.
+#
+# $docker_platform_name, $docker_platform_version, and $package manager need
+# to be set before calling this function.
+install_system_deps () {
+    if [[ "$(id --user)" != "0" ]] && ! has_command sudo ; then
+        echo "Warning: sudo not installed and current user is not root, skipping system dependency installation."
+        return
+    elif ! has_command "$package_manager" ; then
+        echo "Warning: package manager \"$package_manager\" not found, skipping system dependency instllation."
+        return
+    fi
+
+    dockerfile_url="https://raw.githubusercontent.com/apple/swift-docker/main/nightly-main/$docker_platform_name/$docker_platform_version/Dockerfile"
```

Comment:
The system dependencies can change on each release, so I decided to just use the latest set of them to ensure a user can always install the latest swift version successfully. I thought about unioning the oldest releases' dependencies with the newest one's, but that seemed overly complex for an uncommon case.